esphome/components/seeed_mr24hpc1/seeed_mr24hpc1.h

```diff
@@ -0,0 +1,239 @@
+
+  CUSTOM_FUNCTION_QUERY_EXISTENCE_BOUNDARY,
+  CUSTOM_FUNCTION_QUERY_MOTION_BOUNDARY,
+  CUSTOM_FUNCTION_QUERY_EXISTENCE_THRESHOLD,
+  CUSTOM_FUNCTION_QUERY_MOTION_THRESHOLD,
+  CUSTOM_FUNCTION_QUERY_MOTION_TRIGGER_TIME,
+  CUSTOM_FUNCTION_QUERY_MOTION_TO_REST_TIME,
+  CUSTOM_FUNCTION_QUERY_TIME_OF_ENTER_UNMANNED,
+
+  UNDERLY_FUNCTION_QUERY_HUMAN_STATUS,
+  // UNDERLY_FUNCTION_QUERY_SPATIAL_STATIC_VALUE,
+  // UNDERLY_FUNCTION_QUERY_SPATIAL_MOTION_VALUE,
+  // UNDERLY_FUNCTION_QUERY_DISTANCE_OF_STATIC_OBJECT,
+  // UNDERLY_FUNCTION_QUERY_DISTANCE_OF_MOVING_OBJECT,
+  // UNDERLY_FUNCTION_QUERY_TARGET_MOVEMENT_SPEED,
+};
+
+enum {
+  OUTPUT_SWITCH_INIT,
+  OUTPUT_SWTICH_ON,
+  OUTPUT_SWTICH_OFF,
+};
+
+static const std::map<std::string, uint8_t> SCENEMODE_ENUM_TO_INT{
+    {"None", 0x00}, {"Living Room", 0x01}, {"Bedroom", 0x02}, {"Washroom", 0x03}, {"Area Detection", 0x04}};
+
+static const std::map<std::string, uint8_t> UNMANDTIME_ENUM_TO_INT{{"None", 0x00},  {"10s", 0x01},   {"30s", 0x02},
+                                                                   {"1min", 0x03},  {"2min", 0x04},  {"5min", 0x05},
+                                                                   {"10min", 0x06}, {"30min", 0x07}, {"60min", 0x08}};
+
+static const std::map<std::string, uint8_t> BOUNDARY_ENUM_TO_INT{
+    {"0.5m", 0x01}, {"1.0m", 0x02}, {"1.5m", 0x03}, {"2.0m", 0x04}, {"2.5m", 0x05},
+    {"3.0m", 0x06}, {"3.5m", 0x07}, {"4.0m", 0x08}, {"4.5m", 0x09}, {"5.0m", 0x0a},
+};
+
+static const char *const S_SCENE_STR[5] = {"None", "Living Room", "Bedroom", "Washroom", "Area Detection"};
+static const bool S_SOMEONE_EXISTS_STR[2] = {false, true};
+static const char *const S_MOTION_STATUS_STR[3] = {"None", "Motionless", "Active"};
+static const char *const S_KEEP_AWAY_STR[3] = {"None", "Close", "Away"};
+static const char *const S_UNMANNED_TIME_STR[9] = {"None", "10s",   "30s",   "1min", "2min",
+                                                   "5min", "10min", "30min", "60min"};
+static const char *const S_BOUNDARY_STR[10] = {"0.5m", "1.0m", "1.5m", "2.0m", "2.5m",
+                                               "3.0m", "3.5m", "4.0m", "4.5m", "5.0m"};       // uint: m
+static const float S_PRESENCE_OF_DETECTION_RANGE_STR[7] = {0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0};  // uint: m
+
+class MR24HPC1Component : public PollingComponent,
+                          public uart::UARTDevice {  // The class name must be the name defined by text_sensor.py
+#ifdef USE_TEXT_SENSOR
+  SUB_TEXT_SENSOR(heartbeat_state)
+  SUB_TEXT_SENSOR(product_model)
+  SUB_TEXT_SENSOR(product_id)
+  SUB_TEXT_SENSOR(hardware_model)
+  SUB_TEXT_SENSOR(firware_version)
+  SUB_TEXT_SENSOR(keep_away)
+  SUB_TEXT_SENSOR(motion_status)
+  SUB_TEXT_SENSOR(custom_mode_end)
+#endif
+#ifdef USE_BINARY_SENSOR
+  SUB_BINARY_SENSOR(someoneExists)
+#endif
+#ifdef USE_SENSOR
+  SUB_SENSOR(custom_presence_of_detection)
+  SUB_SENSOR(movementSigns)
```

Comment:
variables shoudl always be `lower_snake_case`
```suggestion
  SUB_SENSOR(movement_signs)
```